MekHQ/src/mekhq/campaign/unit/Unit.java

```diff
@@ -4508,8 +4508,8 @@ public void startMothballing(UUID id) {
      * @param isGM A value indicating if the mothball action should
      *             be performed immediately by the GM.
      */
-    public void startMothballing(UUID id, boolean isGM) {
-        if (!isMothballed()) {
+    public void startMothballing(Campaign campaign, UUID id, boolean isGM) {
+        if (!isMothballed() && MekHQ.getMekHQOptions().getSaveMothballState()) {
```

Comment:
Why add campaign as a parameter here? We already have a pointer to the campaign in the Unit class, so it seems unnecessary.

Reply:
Better code style, and unit should not include a pointer to campaign for OOP.

Reply:
Eye of the beholder (I personally find it highly annoying to pass campaign objects around every time I call a method), and not sure object-oriented programming principles recommend one way or the other in terms of keeping pointers to frequently-referenced objects. 

You can make the argument that "startMothballing" doesn't necessarily belong in the Unit class and instead should be moved to some kind of static(?) method in a worker class (e.g. "UnitActionProcessor.startMothballing"), if you want to say that we should avoid mixing data and "business logic" in the same class. As it is, this particular change does nothing to either simplify or improve the code in any meaningful way.

Reply:
I actually agree with you that it is annoying, but it is part of how to handle Objects. Namely, you are supposed to pass the reference in methods when they do not need to be stored with the data for that Object and are entirely external.. and this decoupling may make some future code easier (including a few bits of a longer project I'm working on with person and unit)
The rest is arguable from my experience, and not something I would say.

Reply:
If this isn't going to get a different Campaign object than the unit's, then it shouldn't be added, just adds an argument without benefit.

Reply:
Reverted.